lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -266,9 +280,9 @@ protected function validateRuntimeMetadata($class, $parent)
                 if ( ! $class->discriminatorColumn) {
                     throw MappingException::missingDiscriminatorColumn($class->name);
                 }
-            } else if ($parent && !$class->reflClass->isAbstract() && !in_array($class->name, array_values($class->discriminatorMap))) {
+            } else if (! $class->reflClass->isAbstract() && !in_array($class->name, array_values($class->discriminatorMap))) {
```

Comment:
Removing the check for `$parent` also causes a behavioral change: the root of the inheritance (if non-abstract) must be part of the inheritance.

While this has not been enforced previously, it would have led to runtime problems before.

Now it is enforced/validated when metadata is loaded, therefore the discriminator map must be completely filled, if provided.